src/main/scala/Character.scala

```diff
@@ -0,0 +1,10 @@
+abstract class Character {
```

Comment:
Making the 'Character' trait instead of an abstract class would result in some differences, but many aspects of the functionality would remain the same.
A class can extend only one abstract class but can mix in multiple traits.
Traits cannot have constructor parameters, but since the existing 'Character' class does not include any, it wouldn't cause any problem.